### PR TITLE
[storcli] Add LSI MegaRAID device support

### PR DESCRIPTION
**Pull request by vikas-goel**
Signed-off-by: Vikas Goel <vikas.goel@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: sos/report/plugins/storcli64.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class StorCLI64(Plugin, IndependentPlugin):
+
+    short_desc = 'LSI MegaRAID devices'
+
+    plugin_name = 'storcli64'
+    profiles = ('system', 'storage', 'hardware',)
+    packages = ('storcli',)
+
+    def setup(self):
+        os.environ['PATH'] += os.pathsep + "/opt/MegaRAID/storcli"
```
Comment (TurboTurtle) [inline sos/report/plugins/storcli64.py:22]: Nack to `PATH` modifications - same as in #2610

File: sos/report/plugins/storcli64.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            '/call/eall/sall show all',
+            '/call/sall show all',
+            '/call/vall show all',
+        ]
+
+        logpath = self.get_cmd_output_path()
+
+        self.add_cmd_output([
+            "%s %s" % (cmd, subcmd) for subcmd in subcmds
+        ], runat=logpath)
```
Comment (TurboTurtle) [inline sos/report/plugins/storcli64.py:44]: What's the rationale behind using `runat` here? It's certainly acceptable to do so if needed, but it's very rare to see so I just would like to understand the requirement here.

File: sos/report/plugins/storcli64.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+from sos.report.plugins import Plugin, IndependentPlugin
+import os
+
+
+class StorCLI64(Plugin, IndependentPlugin):
+
+    short_desc = 'LSI MegaRAID devices'
+
+    plugin_name = 'storcli64'
```
Comment (TurboTurtle) [inline sos/report/plugins/storcli64.py:17]: So a while back we updated the `megacli` plugin, and ran into feedback that `storcli` was abandoned even though it was originally intended to _replace_ `megacli`.

Has that changed? Is there actual use/development going on for `storcli`?

If so, I'd suggest a slight modification to the `short_desc`. The `megacli` plugin currently has the same description string, which might confuse users to see two different plugins with the same description.

**Comment by TurboTurtle** [comment]
Also, the Ubuntu 18.04 failure seems unrelated to this PR - it's a not uncommon thing for us to see transient timeouts on 18.04.

I've restarted the job.

File: sos/report/plugins/storcli64.py (hunk 1 above)
Comment (vikas-goel) [reply sos/report/plugins/storcli64.py:22]: Looking forward for your recommendation to fix the path.

File: sos/report/plugins/storcli64.py (hunk 2 above)
Comment (vikas-goel) [reply sos/report/plugins/storcli64.py:44]: The command generates a log file `storcli.log` in current directory. If `runat` is not passed, then the log file is generated in the plugin directory or some other.

File: sos/report/plugins/storcli64.py (hunk 3 above)
Comment (vikas-goel) [reply sos/report/plugins/storcli64.py:17]: Working with the vendor (LSI) recommendation is to use storcli64.
`LSI has created a tool to help manage RAID devices called StorCLI (previously MegaCLI).`

What change in the description would you recommend?

**Comment by vikas-goel** [comment]
@TurboTurtle , thank you for review comments.

**Comment by vikas-goel** [comment]
@TurboTurtle ,
Should I change the plugin name to 'storcli' and still keep the filename `storcli64.py`? Or, would you prefer the plugin name be `storcli64`?

File: sos/report/plugins/storcli64.py (hunk 2 above)
Comment (vikas-goel) [reply sos/report/plugins/storcli64.py:44]: This problem is there with the megacli plugin too. The plugin generates a couple of log files in the current working directory.

**Comment by TurboTurtle** [comment]
I think `storcli` is better here, for both the plugin and filename.

**Comment by vikas-goel** [comment]
Hi @TurboTurtle ,
All suggestions incorporated here.

**Review by TurboTurtle — APPROVED** [review]

**Comment by vikas-goel** [comment]
@pmoravec , please review

**Review by pmoravec — APPROVED** [review]

**Pull request merged**